lib/twitter_cldr/resources/postal_codes_importer.rb

```diff
@@ -3,49 +3,59 @@
 # Copyright 2012 Twitter, Inc
 # http://www.apache.org/licenses/LICENSE-2.0
 
-require 'nokogiri'
-
-require 'twitter_cldr/resources/download'
+require 'rest-client'
+require 'json'
+require 'yaml'
 
 module TwitterCldr
   module Resources
 
     class PostalCodesImporter
 
-      POSTAL_CODES_PATH = File.join('common', 'supplemental', 'postalCodeData.xml')
+      BASE_URL = 'http://i18napis.appspot.com/address/data/'
 
       # Arguments:
       #
-      #   input_path  - path to a directory containing CLDR data
       #   output_path - output directory for generated YAML file
       #
-      def initialize(input_path, output_path)
-        @input_path  = input_path
+      def initialize(output_path)
         @output_path = output_path
       end
 
       def import
-        TwitterCldr::Resources.download_cldr_if_necessary(@input_path)
-
-        doc = File.open(File.join(@input_path, POSTAL_CODES_PATH)) { |file| Nokogiri::XML(file) }
-
-        postal_codes = doc.xpath('//postCodeRegex').inject({}) do |memo, node|
-          memo[node.attr('territoryId').downcase.to_sym] = Regexp.new(node.text); memo
+        File.open(File.join(@output_path, 'postal_codes.yml'), 'w') do |output|
+          output.write(YAML.dump(load))
         end
+      end
 
-        postal_codes = Hash[postal_codes.sort_by(&:first)]
+      private
 
-        postal_codes = postal_codes.each_with_object({}) do |(territory, regex), memo|
-          memo[territory] = {
-            regex: regex,
-            ast: TwitterCldr::Utils::RegexpAst.dump(
-              RegexpAstGenerator.generate(regex.source)
-            )
-          }
+      def load
+        each_territory.each_with_object({}) do |territory, ret|
+          if regex = get_regex_for(territory)
```

Comment:
You can get rid of a level of indentation here by using `next unless regex = get_regex_for(territory)` to get rid of the conditional :)